File: Daily Challanges/2026-02-February/2026-02-26.py
def count_letters_and_numbers(s):
    letter_count = 0
    number_count = 0
    for character in s:
        if character.isalpha():
            letter_count += 1
        elif character.isdigit():
            number_count += 1

    adj1 = "letter" if letter_count == 1 else "letters"
    adj2 = "number" if number_count == 1 else "numbers"

    result = f"The string has {letter_count} {adj1} and {number_count} {adj2}."
    return result

test1 = count_letters_and_numbers("helloworld123")
test2 = count_letters_and_numbers("Catch 22")
test3 = count_letters_and_numbers("A1!")
test4 = count_letters_and_numbers("12345")
test5 = count_letters_and_numbers("password")

check1 = "The string has 10 letters and 3 numbers."
check2 = "The string has 5 letters and 2 numbers."
check3 = "The string has 1 letter and 1 number."
check4 = "The string has 0 letters and 5 numbers."
check5 = "The string has 8 letters and 0 numbers."

print(f"Test1: {test1 == check1}")
print(f"Test2: {test2 == check2}")
print(f"Test3: {test3 == check3}")
print(f"Test4: {test4 == check4}")
print(f"Test5: {test5 == check5}")

File: Daily Challanges/2026-02-February/2026-02-26.js
function countLettersAndNumbers(str) {
  let letterCount = 0;
  let numberCount = 0;
  for(const character of str){
    if(/^[a-zA-Z]$/.test(character)){letterCount++}
    if(/^[0-9]$/.test(character)){numberCount++}
  }
  const adj1 = letterCount === 1 ? "letter" : "letters";
  const adj2 = numberCount === 1 ? "number" : "numbers";
  const result = `The string has ${letterCount} ${adj1} and ${numberCount} ${adj2}.`
  return result;
}

const test1 = countLettersAndNumbers("helloworld123");
const test2 = countLettersAndNumbers("Catch 22");
const test3 = countLettersAndNumbers("A1!");
const test4 = countLettersAndNumbers("12345");
const test5 = countLettersAndNumbers("password");

const check1 = "The string has 10 letters and 3 numbers.";
const check2 = "The string has 5 letters and 2 numbers.";
const check3 = "The string has 1 letter and 1 number.";
const check4 = "The string has 0 letters and 5 numbers.";
const check5 = "The string has 8 letters and 0 numbers.";

console.log(`Test1: ${test1 === check1}`);
console.log(`Test2: ${test2 === check2}`);
console.log(`Test3: ${test3 === check3}`);
console.log(`Test4: ${test4 === check4}`);
console.log(`Test5: ${test5 === check5}`);